optaplanner-examples/src/test/java/org/optaplanner/examples/nqueens/integration/constructionheuristic/ConstructionHeuristicTest.java

```diff
@@ -0,0 +1,201 @@
+import org.optaplanner.examples.nqueens.domain.Column;
+import org.optaplanner.examples.nqueens.domain.NQueens;
+import org.optaplanner.examples.nqueens.domain.Queen;
+import org.optaplanner.examples.nqueens.domain.Row;
+import org.optaplanner.examples.nqueens.integration.util.QueenCoordinates;
+import org.optaplanner.examples.nqueens.integration.util.QueenCoordinatesStepListener;
+import org.optaplanner.examples.nqueens.solver.score.NQueensAdvancedIncrementalScoreCalculator;
+
+import java.util.*;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+@RunWith(Parameterized.class)
+public class ConstructionHeuristicTest {
+
+    private final ConstructionHeuristicType constructionHeuristicType;
+    private final EntitySorterManner entitySorterManner;
+    private final ValueSorterManner valueSorterManner;
+    private final List<QueenCoordinates> expectedCoordinates;
+
+    public ConstructionHeuristicTest(ConstructionHeuristicType constructionHeuristicType,
+                                     EntitySorterManner entitySorterManner, ValueSorterManner valueSorterManner,
+                                     List<QueenCoordinates> expectedCoordinates) {
+        this.constructionHeuristicType = constructionHeuristicType;
+        this.entitySorterManner = entitySorterManner;
+        this.valueSorterManner = valueSorterManner;
+        this.expectedCoordinates = expectedCoordinates;
+    }
+
+    @Test
+    public void testConstructionHeuristics() {
+        SolverConfig config = new SolverConfig();
+        config.setEnvironmentMode(EnvironmentMode.REPRODUCIBLE);
+        config.setSolutionClass(NQueens.class);
+        config.setEntityClassList(Collections.<Class<?>>singletonList(Queen.class));
+
+        config.setScoreDirectorFactoryConfig(new ScoreDirectorFactoryConfig());
+        config.getScoreDirectorFactoryConfig().setScoreDefinitionType(ScoreDefinitionType.SIMPLE);
+        config.getScoreDirectorFactoryConfig()
+                .setIncrementalScoreCalculatorClass(NQueensAdvancedIncrementalScoreCalculator.class);
+
+        config.setTerminationConfig(new TerminationConfig());
+        config.getTerminationConfig().setBestScoreLimit("0");
+
+        ConstructionHeuristicPhaseConfig chConfig = new ConstructionHeuristicPhaseConfig();
+        chConfig.setConstructionHeuristicType(constructionHeuristicType);
+        chConfig.setEntitySorterManner(entitySorterManner);
+        chConfig.setValueSorterManner(valueSorterManner);
+        config.setPhaseConfigList(Collections.<PhaseConfig>singletonList(chConfig));
+
+        NQueens solution = new NQueens();
+        List<Queen> entities = new ArrayList<Queen>();
+        int dimension = 8;
+        List<Row> rowList = new ArrayList<Row>();
+        List<Column> columnList = new ArrayList<Column>();
+        for (int i = 0; i < dimension; i++) {
+            Column column = new Column();
+            column.setIndex(i);
```

Comment:
Id's aren't set on Column, Row or Queen. The id's are used in the rules. That's dangerous.

This code should be replaced by a call to NQueensGenerator.createNQueens().